### PR TITLE
SCP-2325: How to export a script

### PR DESCRIPTION
**Pull request by michaelpj**
Extend the basic validators tutorial to talk about the typed interface,
also add a short how-to about how to export a script. It has a big TODO
though, since we don't have the envelope support yet, but that should be
easy to add later.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: doc/plutus/tutorials/BasicValidators.hs [hunk 1]
```diff
@@ -77,9 +84,36 @@ validatePayment _ _ ctx = check $ case fromData ctx of
...
+dateValidatorHash = scriptHash dateInstance
+
+dateValidator :: Validator
+dateValidator = validatorScript dateInstance
+-- BLOCK8
+serializedDateValidator :: BSL.ByteString
+serializedDateValidator = serialise dateValidator
+
+main :: IO ()
+main = print serializedDateValidator
```
Comment (j-mueller) [inline doc/plutus/tutorials/BasicValidators.hs:118]: very minimal ;)

**Review by j-mueller — APPROVED** [review]
Just a thought: We could add a way to obtain a `ScriptInstance` from a serialised validator, to help with the stable hashes problem.

File: doc/plutus/tutorials/BasicValidators.hs (hunk 1 above)
Comment (michaelpj) [reply doc/plutus/tutorials/BasicValidators.hs:118]: Yeah, there wasn't that much to say, but I wanted to go to the end...

**Comment by michaelpj** [comment]
> Just a thought: We could add a way to obtain a ScriptInstance from a serialised validator, to help with the stable hashes problem.

I think there's already a `Validator -> ScriptInstance Any` function, but we could have an unsafe version that lets you pick the `a`, and I think that would be okay.

**Pull request merged**